modules/capture-video/src/main/native/linux/src/capture.c

```diff
@@ -0,0 +1,214 @@
+// INSPIRED BY: 
+/* V4L2 video picture grabber
+   Copyright (C) 2009 Mauro Carvalho Chehab <mchehab@infradead.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   Modified by Derek Molloy (www.derekmolloy.ie)
+   Modified to change resolution details and set paths for the Beaglebone.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <libudev.h>
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+#include <linux/videodev2.h>
+#include <libv4l2.h>
+#include <unistd.h>
+
+#include "jni.h"
+
+JNIEnv *jEnv = NULL;
+jclass jCaptureVideoServiceClass;
+jmethodID jCaptureVideoService_setResult = 0;
+
+
+int active = 0;
+
+#define CLEAR(x) memset(&(x), 0, sizeof(x))
+
+struct buffer {
+        void   *start;
+        size_t length;
+};
+
+static void xioctl(int fh, int request, void *arg) {
+        int r;
+        do {
+                r = v4l2_ioctl(fh, request, arg);
+        } while (r == -1 && ((errno == EINTR) || (errno == EAGAIN)));
+        if (r == -1) {
+                fprintf(stderr, "error %d, %s\n", errno, strerror(errno));
+                exit(EXIT_FAILURE);
+        }
+}
+
+int initializeGrabber() {
+}
+
+void sendPicturesResult(int width, int height, int format, uint8_t* data, size_t len) {
+    fprintf(stderr, "send pic, format = %d, len = %ld\n", format,len);
+    JNIEnv *env = jEnv;
+    jbyteArray picByteArray = (*env)->NewByteArray(env, len);
+    (*env)->SetByteArrayRegion(env, picByteArray, 0, len, data);
+    if ((*env)->ExceptionCheck(env)) {
+        (*env)->ExceptionDescribe(env);
+        (*env)->ExceptionClear(env);
+        return;
+    }
+    fprintf(stderr, "send pic2, format = %d\n", format);
+    (*env)->CallStaticVoidMethod(env, jCaptureVideoServiceClass, jCaptureVideoService_setResult, width, height, format, picByteArray);
+    fprintf(stderr, "send pic3, format = %d\n", format);
+    (*env)->DeleteLocalRef(env, picByteArray);
+    fprintf(stderr, "send pic4, format = %d\n", format);
+}
+
+void rgbToRgba(char* rgba, const char* rgb, const int count) {
+    if(count==0) return;
+    for(int i=count; --i; rgba+=4, rgb+=3) {
+        *(uint32_t*)(void*)rgba = *(const uint32_t*)(const void*)rgb;
+    }
+    for(int j=0; j<3; ++j) {
```

Comment:
shouldn't be this `for` nested in the above? 

Also, for the alpha channel:
```
rgba[3] = 0xFF;
```
?

Reply:
I'll check this later. The whole linux file needs a rework, but I wanted to make sure we can deploy first.